resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ResourceTypeHelperSuite.scala

```diff
@@ -0,0 +1,153 @@
+    assume(ResourceTypeHelper.isYarnResourceTypesAvailable())
+    TestYarnResourceTypeHelper.initializeResourceTypes(List())
+
+    val resourceTypes = Map(CUSTOM_RES_1 -> "123")
+
+    val thrown = intercept[IllegalArgumentException] {
+      ResourceTypeHelper.setResourceInfoFromResourceTypes(resourceTypes, null)
+    }
+    thrown.getMessage should equal ("Resource parameter should not be null!")
+  }
+
+  test("resource type with valid value and invalid unit") {
+    assume(ResourceTypeHelper.isYarnResourceTypesAvailable())
+    TestYarnResourceTypeHelper.initializeResourceTypes(List(CUSTOM_RES_1))
+
+    val resourceTypes = Map(CUSTOM_RES_1 -> "123ppp")
+    val resource = createAResource
+
+    val thrown = intercept[IllegalArgumentException] {
+      ResourceTypeHelper.setResourceInfoFromResourceTypes(resourceTypes, resource)
+    }
+    thrown.getMessage should fullyMatch regex
+      """Unknown unit 'ppp'\. Known units are \[.*\]"""
+  }
+
+  test("resource type with valid value and without unit") {
+    assume(ResourceTypeHelper.isYarnResourceTypesAvailable())
+    TestYarnResourceTypeHelper.initializeResourceTypes(List(CUSTOM_RES_1))
+
+    val resourceTypes = Map(CUSTOM_RES_1 -> "123")
+    val resource = createAResource
+
+    ResourceTypeHelper.setResourceInfoFromResourceTypes(resourceTypes, resource)
+    val customResource: ResourceInformation = TestYarnResourceTypeHelper
+      .getResourceInformationByName(resource, CUSTOM_RES_1)
+    customResource.name should equal (CUSTOM_RES_1)
+    customResource.value should be (123)
+    customResource.units should be ("")
+  }
+
+  test("resource type with valid value and unit") {
+    assume(ResourceTypeHelper.isYarnResourceTypesAvailable())
+    TestYarnResourceTypeHelper.initializeResourceTypes(List(CUSTOM_RES_1))
+
+    val resourceTypes = Map(CUSTOM_RES_1 -> "2g")
+    val resource = createAResource
+
+    ResourceTypeHelper.setResourceInfoFromResourceTypes(resourceTypes, resource)
+    val customResource: ResourceInformation = TestYarnResourceTypeHelper
+      .getResourceInformationByName(resource, CUSTOM_RES_1)
+    customResource.name should equal (CUSTOM_RES_1)
+    customResource.value should be (2)
+    customResource.units should be ("G")
+  }
+
+  test("two resource types with valid values and units") {
+    assume(ResourceTypeHelper.isYarnResourceTypesAvailable())
+    TestYarnResourceTypeHelper.initializeResourceTypes(List(CUSTOM_RES_1, CUSTOM_RES_2))
+
+    val resourceTypes = Map(
+      CUSTOM_RES_1 -> "123m",
+      CUSTOM_RES_2 -> "10G"
+    )
+    val resource = createAResource
+
+    ResourceTypeHelper.setResourceInfoFromResourceTypes(resourceTypes, resource)
+    val customResource1: ResourceInformation = TestYarnResourceTypeHelper
+      .getResourceInformationByName(resource, CUSTOM_RES_1)
+    customResource1.name should equal (CUSTOM_RES_1)
+    customResource1.value should be (123)
+    customResource1.units should be ("m")
```

Comment:
This line fails for me. It should be `customResource1.units should be ("M")`, right?

Reply:
good catch, fixed